src/main/java/edu/cornell/kfs/sys/batch/DocumentReindexStep.java

```diff
@@ -4,56 +4,70 @@
-
-	    try {
-	    	BufferedReader reader = new BufferedReader(new FileReader(f));
-
-	    	String line = null;
-	    	while ((line=reader.readLine()) != null) {
-	    		docIds.add(line);   	
-	    	}
-	    } catch (IOException ioe) {
-	    	ioe.printStackTrace();
-	    	return false;
-	    }
-		for (Iterator<String> it = docIds.iterator(); it.hasNext(); ) {
-			String documentId = it.next();
-			
-			try {
-			    LOG.info("execute, indexing document " + documentId);
-			    documentAttributeIndexingQueue.indexDocument(documentId);
-			} catch (Exception e) {
-			    LOG.error("execute, had an error indexing docuemnt " + documentId, e);
-			}
-		}
-		
-		addTimeStampToFileName(f, fileName, stagingDirectory);
-		return true;
-	}
-
-	public void setStagingDirectory(String stagingDirectory) {
-		this.stagingDirectory = stagingDirectory;
-	}
+
+    private static final Logger LOG = LogManager.getLogger();
+
+    private DocumentAttributeIndexingQueue documentAttributeIndexingQueue;
+    private String stagingDirectory;
+
+    public boolean execute(String jobName, Date jobRunDate) throws InterruptedException {
+        try {
+            File documentReindexFile = new File(stagingDirectory + File.separator + getDocumentReindexFilename());
+            List<String> documentIds = getDocumentIdsToProcess(documentReindexFile);
+            reindexDocuments(documentIds);
+            addTimeStampToFileName(documentReindexFile, getDocumentReindexFilename(), stagingDirectory);
+        } catch (Exception e) {
+            LOG.error("execute, Unexpected error occurred while indexing documents", e);
+            throw new RuntimeException(e);
+        }
+        return true;
+    }
+
+    private String getDocumentReindexFilename() {
+        return CUKFSConstants.DOCUMENT_REINDEX_FILE_NAME_PREFIX + CUKFSConstants.DOCUMENT_REINDEX_FILE_NAME_EXTENSION;
+    }
+
+    private List<String> getDocumentIdsToProcess(File documentReindexFile) throws IOException {
+        LOG.info("getDocumentIdsToProcess, Reading document IDs to be processed for reindexing...");
+        try (FileReader fileReader = new FileReader(documentReindexFile, StandardCharsets.UTF_8);
+                BufferedReader bufferedReader = new BufferedReader(fileReader);
+        ) {
+            List<String> documentIds = bufferedReader.lines().collect(Collectors.toUnmodifiableList());
+            LOG.info("getDocumentIdsToProcess, Found " + documentIds.size() + " documents to reindex");
+            return documentIds;
+        }
+    }
+
+    private void reindexDocuments(List<String> documentIds) {
+        for (String documentId : documentIds) {
+            try {
+                LOG.info("reindexDocuments, Indexing document " + documentId);
+                documentAttributeIndexingQueue.indexDocument(documentId);
+            } catch (Exception e) {
+                LOG.error("reindexDocuments, Unexpected error occurred while indexing document " + documentId, e);
```

Comment:
The exception generated here is not re-thrown. This will allow processing to be attempt for all documents listed in the file and it is good that the error is logged for each document that failed. 
Asking if more should be done to "alert" that there is an issue with a particular document so that these failures don't repeatedly happen and "get lost"?

Reply:
Apart from triggering a failure of the batch job as a whole, we could have the job send an email or generate a report file to indicate which documents failed to reindex (if any). However, I'm not certain whether it would be worth spending the effort to implement something like that at the present time; it might make more sense to have it as a future enhancement. Let me know what you think.

Reply:
Having this as a future enhancement makes sense. That will allow us to see what the behavior is like before we invest any more effort.

Reply:
Sounds good, I'll create a follow-up enhancement user story accordingly. Thanks for the input!